src/client/Microsoft.Identity.Client/ManagedIdentity/V2/PersistentCertificateStore.cs

```diff
@@ -0,0 +1,259 @@
+                {
+                    try
+                    {
+                        if (!FriendlyNameCodec.TryDecode(c.FriendlyName, out var alias, out var epBase))
+                            continue;
+                        if (!StringComparer.Ordinal.Equals(alias, aliasCacheKey))
+                            continue;
+
+                        // 24h+ remaining
+                        if (c.NotAfter.ToUniversalTime() <= DateTime.UtcNow + CertificateCacheEntry.MinRemainingLifetime)
+                            continue;
+
+                        if (!c.HasPrivateKey)
+                        {
+                            logger?.Verbose(() => "[PersistentCert] Candidate skipped: no private key.");
+                            continue;
+                        }
+
+                        if (c.NotAfter > bestNotAfter)
+                        {
+                            best?.Dispose();
+                            best = new X509Certificate2(c); // caller-owned clone (preserves private key link)
+                            bestEndpoint = epBase;
+                            bestNotAfter = c.NotAfter;
+                        }
+                    }
+                    finally
+                    {
+                        c.Dispose();
+                    }
+                }
+
+                if (best != null)
+                {
+                    // CN (GUID) → canonical client_id
+                    string cn = null;
+                    try
+                    { cn = best.GetNameInfo(X509NameType.SimpleName, false); }
+                    catch { }
+                    if (!Guid.TryParse(cn, out var g))
+                    {
+                        best.Dispose();
+                        logger?.Verbose(() => "[PersistentCert] Selected entry CN is not a GUID; skipping.");
+                        return false;
+                    }
+
+                    value = new CertificateCacheValue(best, bestEndpoint, g.ToString("D"));
+                    logger?.Verbose(() => "[PersistentCert] Reused certificate from CurrentUser/My.");
+                    return true;
+                }
+            }
+            catch (Exception ex)
+            {
+                logger?.Verbose(() => "[PersistentCert] Store lookup failed: " + ex.Message);
+            }
+
+            return false;
+        }
+
+        public static void TryPersist(
+            string aliasCacheKey,
+            X509Certificate2 cert,
+            string endpointBase,
+            string clientId,           // unused; CN is the source of truth on read
+            ILoggerAdapter logger = null)
+        {
+            if (!DesktopOsHelper.IsWindows() || cert == null)
+                return;
+
+            // We only persist certs that can actually be used for mTLS later.
+            if (!cert.HasPrivateKey)
+            {
+                logger?.Verbose(() => "[PersistentCert] Not persisting: certificate has no private key.");
```

Comment:
Seems like a warning to me? Or even an exception? Why would you persist certs without private key?